core/src/ibc/connection_03/manager.rs

```diff
@@ -135,13 +174,19 @@ impl<'a> Manager<'a> {
         &mut self,
         connection: &ConnectionEnd,
         proof_height: u64,
-        proof: CommitmentProof,
+        proof: Vec<u8>,
         connection_identifier: Identifier,
         connection_end: &ConnectionEnd,
     ) -> bool {
+        let proof_dec: CommitmentProof = if let Ok(proof) = rlp::decode(&proof) {
```

Comment:
decoding should happen only in the client.

Reply:
I think that the whole `verify_connection_state` function will be moved to the light client soon.

Reply:
Ah ok. You just wanted to keep that consistent. I'll move it soon, so never mind.